src/isomorphism/lad.c

```diff
@@ -110,7 +110,7 @@ static int igraph_i_lad_createGraph(const igraph_t *igraph, Tgraph* graph) {
 
     graph->nbVertices = no_of_nodes;
 
-    IGRAPH_CHECK(igraph_adjlist_init(igraph, &graph->succ, IGRAPH_OUT));
+    IGRAPH_CHECK(igraph_adjlist_init(igraph, &graph->succ, IGRAPH_OUT, IGRAPH_LOOPS_ONCE, IGRAPH_MULTIPLE));
```

Comment:
@ntamas There's a bit of a problem here.

Previously, when this was effectively LOOPS_TWICE, it caused the non-simple graph error to trigger a few lines below. Now that error does not trigger on graphs that have self-loops but no multi-edges.

However, now, a few lines above, the computation of `nbSucc` using `igraph_degree` is incorrect because that line effectively double-counts loops.  I fixed that and tried many examples to see if LAD can handle self-loops. I did not so far encounter any incorrect result. That's great, it looks like we just fixed a problem and gain support for loopy graphs with LAD! But: with this are taking a risk. I only verified that it works empirically ... 

So, what do we do about this?

 1. We detect loopy graphs and report an error
 2. We fix the `igraph_degree` line and let LAD work with loopy graphs

I am cautiously leaning towards solution 2.

Reply:
I just committed proposal 2 to show what I mean.

Reply:
Hmmm, okay, I should have probably written `IGRAPH_LOOPS_TWICE`; I was confused by `IGRAPH_OUT`, which works like `IGRAPH_LOOPS_ONCE` for directed graphs but is ignored (and treated as `IGRAPH_ALL`) for undirected graphs. I don't know off the top of my head whether LAD supports directed graphs; if it doesn't, then I should have written `IGRAPH_LOOPS_TWICE`, shouldn't I?

Reply:
LAD supports directed graphs, but it is not optimized for them. This is why the website say "for non directed graphs": http://perso.citi-lab.fr/csolnon/LAD.html

With the commit I pushed, it seems to handle self-loops fine. I just did a bit of random fuzzing, and it does not seem to fail to detect isomorphism, nor does it return a wrong mapping. Why I did not (could not) test with the fuzzer was that it wouldn't give false negatives (i.e. say "not isomorphic" when in fact it is). But I also did quite a bit manual testing, and it seems fine.